pkgs/development/tools/godot/default.nix

```diff
@@ -19,30 +19,48 @@
 , libGLU
 , zlib
 , yasm
-, withUdev ? true
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self:
+{
   pname = "godot";
   version = "3.4.4";
 
+  name = "${self.pname}-${self.version}";
+
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
-    sha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+    rev = self.passthru.githubRev;
+    sha256 = self.passthru.githubSha256;
   };
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [
+  passthru = {
+    githubRev = "${self.version}-stable";
+    githubSha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+
+    sconsFlagSet = {
+      target = "release_debug";
+      platform = "x11";
+      tools = true;
+      touch = libXi != null;
+      pulseaudio = false;
+      udev = true;
+    };
```

Comment:
This is in passthru rather than outside of it because it's not directly used within the phases.  Instead it is an overridable attrset from which the `sconsFlags` string is generated, and that string is consumed by the scons phases.